TLM/TLM/Util/RoundaboutMassEdit.cs

```diff
@@ -173,23 +173,29 @@ public class RoundaboutMassEdit {
         /// </summary>
         /// <param name="segmentId"></param>
         /// <returns></returns>
-        public bool FixRabout(ushort initialSegmentId) {
-            bool isRAbout = TraverseLoop(initialSegmentId, out _);
-            if (!isRAbout) {
+        public bool FixRoundabout(ushort initialSegmentId) {
+            bool isRoundabout = TraverseLoop(initialSegmentId, out var segmentList);
+            if (!isRoundabout) {
                 Log._Debug($"segment {initialSegmentId} not a roundabout.");
                 return false;
             }
             int count = segmentList.Count;
             Log._Debug($"\n segmentId={initialSegmentId} seglist.count={count}\n");
 
+            FixRoundabout(segmentList);
+            return true;
+        }
+
+        public void FixRoundabout(List<ushort> segmentList) {
+            this.segmentList = segmentList;
+            int count = segmentList?.Count ?? 0;
```

Comment:
Do you really want `NullReferenceException` here? 
Pretty unfortunate name for parameter, btw

Reply:
@kianzarrin it's still not resolved!
Case is simple. Why you are afraid of `null` in `segmentList`, but the same null you are setting to `this.segmentList` field and inside other methods you use `this.segmentList.Clear()` without bothering with null check?

I don't understand why did you overwritten field value but still use param for next operations. I think that method does to much things at once